vmm/src/vm.rs

```diff
@@ -929,39 +985,45 @@ impl Vm {
         Ok(())
     }
 
+    // Load image as pe file first, if fails, load it as plain file.
+    #[cfg(target_arch = "aarch64")]
+    fn load_pe_or(
```

Comment:
I think we don't need to add yet another function. See the comment on `load_payload()`.

And some question regarding this code changes: 
- If I assign a kernel binary to `--firmware`, does the guest boots from that kernel? If yes, that's not expected.
- If I assign a kernel to `--firmware` and the same kernel to `--kernel`, does the guest boots?